Produlytics/ui/front-end/src/app/model/http-interceptor/http-interceptor.service.ts

```diff
@@ -54,18 +54,6 @@ export class XhrInterceptor implements HttpInterceptor {
       req.headers.set('X-XSRF-TOKEN', csrfToken);
```

Comment:
@MicheGit sicuro che questo modifichi `req.header` e non crei invece un clone con la modifica?

Reply:
certo

https://developer.mozilla.org/en-US/docs/Web/API/Headers/set

tieni a mente che se non modificasse `req.header` le `PUT` non funzionerebbero

Reply:
Ma `req` è un `HttpRequest` di Angular, il cui campo `headers` ha tipo `HttpHeaders` e non `Headers`, e la documentazione di `HttpHeaders.set` dice che crea un clone https://angular.io/api/common/http/HttpHeaders#set

Inoltre anche la documentazione di `HttpInterceptor` di angular ha un esempio in cui il valore di ritorno di `set` viene utilizzato per creare una nuova richiesta https://angular.io/guide/http#setting-default-headers

Reply:
In aggiunta al mio precedente commento, anche se `set` modificasse il valore esistente di `headers`, è comunque incoerente fare una volta un clone e una volta no.

Reply:
Ho appena provato a far partire il docker e a guardare le richieste che vengono effettuate attraverso gli strumenti sviluppatore del browser. `X-Requested-With: XMLHttpRequest` è correttamente impostato ma `X-XSRF-TOKEN` no.